File: Programmers/level 2/문자열압축.js
function solution(s) {
    let answer = s;
    const slen=s.length;
    //최대 압축할 수 있는 길이는 s의 길이의 절반
    //i개 단위로 압축
    for (let i=1; i<=slen/2; i++){
        let cnt=1;
        let temp="";
        for (let j=0; j<slen; j+=i){
            //앞 뒤 단위로 자른 문자열 비교
            let current=s.substring(j, j+i);
            let next=s.substring(j+i, j+i+i);
            if (current===next) cnt++;
            else {
                if (cnt>1) temp=temp+cnt+current;
                else {
                    temp+=current;
                }
                cnt=1;
            }
        }
        if (answer.length>temp.length) answer=temp;
    }
    return answer.length;
}

File: Programmers/level 2/문자열 압축.cpp
#include <string>
#include <vector>
using namespace std;

int solution(string s) {
    int answer = s.length();
    //최대로 압축할 수 있는 길이는 1~s.length()/2만큼!
    for (int i=1; i<=s.length()/2; i++) {
        int cnt=1;
        string temp="";
        string cur=s.substr(0, i);
        for (int j=i; j<s.length(); j+=i) {
            string next=s.substr(j, i);
            if (cur==next) cnt++;
            else {
               if (cnt>1) temp+=to_string(cnt);
                temp+=cur;
                cnt=1; 
                cur=next;
            }
        }
        if (cnt>1) temp+=to_string(cnt); 
        temp+=cur;
        if (answer>temp.length()) answer=temp.length();
    }
    return answer;
}
